### PR TITLE
Deprecate QueueAckLevel

### PR DESCRIPTION
**Pull request by yycptt**
**What changed?**
- Deprecate QueueAckLevel and related compatibility code

**Why?**
- In 1.21 all logic are already using queueStates field in shardInfo. QueueAckLevel is no longer needed in 1.22.

**How did you test it?**
- Existing test

**Potential risks**

**Is hotfix candidate?**
- No.

### CODE REVIEW

File: service/history/replication/task_processor_test.go [hunk 1]
```diff
@@ -114,13 +113,13 @@ func (s *taskProcessorSuite) SetupTest() {
...
-					},
-				},
-			},
+			// QueueAckLevels: map[int32]*persistencespb.QueueAckLevel{
+			// 	tasks.CategoryReplication.ID(): {
+			// 		ClusterAckLevel: map[string]int64{
+			// 			cluster.TestAlternativeClusterName: persistence.EmptyQueueMessageID,
+			// 		},
+			// 	},
+			// },
```
Comment (yujieli-temporal) [inline service/history/replication/task_processor_test.go:122]: qq, why comment but not remove?

**Review by yujieli-temporal — APPROVED** [review]

File: service/history/replication/task_processor_test.go (hunk 1 above)
Comment (yiminc) [reply service/history/replication/task_processor_test.go:122]: please remove

**Review by yiminc — APPROVED** [review]

File: service/history/replication/task_processor_test.go (hunk 1 above)
Comment (yycptt) [reply service/history/replication/task_processor_test.go:122]: oops, missed that. Will remove.

**Review by wxing1292 — APPROVED** [review]
LGTM

**Pull request merged**